Mathlib/Topology/Algebra/Constructions.lean

```diff
@@ -180,4 +180,32 @@ theorem continuous_coe_inv : Continuous (fun u => ↑u⁻¹ : Mˣ → M) :=
 #align units.continuous_coe_inv Units.continuous_coe_inv
 #align add_units.continuous_coe_neg AddUnits.continuous_coe_neg
 
+lemma top_eq_if_cont_inv' {α : Type*} [TopologicalSpace α] [Monoid α]
```

Comment:
I think we should try to avoid working with different topologies on a single space. Would the following lemmas cover your use case?
```lean
@[to_additive]
protected lemma continuous_inv : Continuous (fun u ↦ u⁻¹ : Mˣ → Mˣ) := by
  rw [Units.continuous_iff]
  exact ⟨continuous_coe_inv, continuous_val⟩

@[to_additive (attr := simps!)]
protected def homeomorph_inv : Mˣ ≃ₜ Mˣ :=
  { left_inv := fun _ ↦ rfl,
    right_inv := fun _ ↦ rfl,
    continuous_toFun := Units.continuous_inv,
    continuous_invFun := Units.continuous_inv }

lemma inducing_coe_iff : Inducing (coeHom M) ↔ Inducing (fun u ↦ ↑u⁻¹ : Mˣ → M) :=
  ⟨fun h ↦ h.comp (Units.homeomorph_inv.inducing),
   fun h ↦ h.comp (Units.homeomorph_inv.inducing)⟩

lemma embedding_coe_iff : Embedding (coeHom M) ↔ Embedding (fun u ↦ ↑u⁻¹ : Mˣ → M) :=
  ⟨fun h ↦ h.comp (Units.homeomorph_inv.embedding),
   fun h ↦ h.comp (Units.homeomorph_inv.embedding)⟩
```